tests/test_statistics.py

```diff
@@ -99,6 +101,30 @@ def test_max_min_error(func):
         getattr(dpnp, func)(ia, initial=6)
 
 
+@pytest.mark.parametrize("func", ["nanmax", "nanmin"])
+@pytest.mark.parametrize("dtype", get_all_dtypes(no_bool=True))
+def test_nanmax_nanmin_no_NaN(func, dtype):
+    a = numpy.arange(768, dtype=dtype).reshape((4, 4, 6, 8))
+    ia = dpnp.array(a)
+
+    np_res = getattr(numpy, func)(a, axis=0)
+    dpnp_res = getattr(dpnp, func)(ia, axis=0)
+    assert_dtype_allclose(dpnp_res, np_res)
+
+
+@pytest.mark.parametrize("func", ["nanmax", "nanmin"])
+def test_nanmax_nanmin_all_NaN(func):
+    a = numpy.arange(12, dtype=numpy.float32).reshape((2, 2, 3))
+    a[:, :, 2] = numpy.nan
+    ia = dpnp.array(a)
+
+    with warnings.catch_warnings():
```

Comment:
that doesn't allow to check that dpnp and numpy raise the same warning, I'd suggest something like:
```python
def test_nanmax_nanmin_all_nan(recwarn, func):
    a = numpy.arange(12, dtype=numpy.float32).reshape((2, 2, 3))
    a[:, :, 2] = numpy.nan
    ia = dpnp.array(a)

    np_res = getattr(numpy, func)(a, axis=0)
    dpnp_res = getattr(dpnp, func)(ia, axis=0)
    assert_dtype_allclose(dpnp_res, np_res)

    assert len(recwarn) == 2
    assert all("All-NaN slice encountered" in str(r.message) for r in recwarn)
    assert all(r.category is RuntimeWarning for r in recwarn)
```
through `recwarn` pytest fixture